src/v/storage/segment_appender.cc

```diff
@@ -445,66 +446,113 @@ void segment_appender::dispatch_background_head_write() {
 
     const size_t start_offset = ss::align_down<size_t>(
       _committed_offset, _head->alignment());
-    const size_t expected = _head->dma_size();
-    const char* src = _head->dma_ptr();
-    vassert(
-      expected <= _chunk_size,
-      "Writes can be at most a full segment. Expected {}, attempted write: {}",
-      _chunk_size,
-      expected);
+
     // accounting synchronously
+    const auto prior_co = _committed_offset;
     _committed_offset += _head->bytes_pending();
     _bytes_flush_pending -= _head->bytes_pending();
+
+    inflight_write entry{
+      .chunk = _head,
+      .chunk_begin = _head->pending_aligned_begin(),
+      .chunk_end = _head->pending_aligned_end(),
+      .file_start_offset = start_offset,
+      .committed_offset = _committed_offset};
+
     // background write
     _head->flush();
+
+    entry.full = _head->is_full();
```

Comment:
nit: you could init `entry.full` when you construct `entry` on line 455.

Reply:
@VladLazar - no, while I would have also liked this it goes after because `_head->flush()` modifies head which affects the result of `is_full()`: we need to know if the head is full _after_ the flush() is applied.

Reply:
Maybe I'm missing something, but I don't see how `flush` impacts the result of `is_full`:

```cpp
// segment_appender_chunk.cc
bool is_full() const { return _pos == _chunk_size; }
void flush() { _flushed_pos = _pos; }
 ```

Reply:
@VladLazar - no you aren't missing anything, I had wrongly thought that the flush affected `is_full` but it doesn't: it's the earlier `_head->append` that adjusts `_pos` so it's safe to init it at the same time as the rest, and I"ll do so.